packages/netlify-cms-backend-github/src/API.js

```diff
@@ -1051,47 +1024,17 @@ export default class API {
     );
   }
 
-  updateTree(sha, path, fileTree) {
-    return this.getTree(sha).then(tree => {
-      let obj;
-      let filename;
-      let fileOrDir;
-      const updates = [];
-      const added = {};
-
-      for (let i = 0, len = tree.tree.length; i < len; i++) {
-        obj = tree.tree[i];
-        if ((fileOrDir = fileTree[obj.path])) {
-          // eslint-disable-line no-cond-assign
-          added[obj.path] = true;
-          if (fileOrDir.file) {
-            updates.push({ path: obj.path, mode: obj.mode, type: obj.type, sha: fileOrDir.sha });
-          } else {
-            updates.push(this.updateTree(obj.sha, obj.path, fileOrDir));
-          }
-        }
-      }
-      for (filename in fileTree) {
-        fileOrDir = fileTree[filename];
-        if (added[filename]) {
-          continue;
-        }
-        updates.push(
-          fileOrDir.file
-            ? { path: filename, mode: '100644', type: 'blob', sha: fileOrDir.sha }
-            : this.updateTree(null, filename, fileOrDir),
-        );
-      }
-      return Promise.all(updates)
-        .then(tree => this.createTree(sha, tree))
-        .then(response => ({
-          path,
-          mode: '040000',
-          type: 'tree',
-          sha: response.sha,
-          parentSha: sha,
-        }));
-    });
+  async updateTree(sha, files) {
```

Comment:
Used nested file paths to update the tree instead of recursion (@erquhart figured it).
This is also the only way we got file deletion to work.